docs/contributor-guide/developer-guide/authentication-providers.md

```diff
@@ -8,6 +8,89 @@ This page will go over the implementation and configuration steps for currently
 
 ### Azure AD (OIDC)
 
+Below you'll find instructions on how to use Azure Active Directory to authenticate CiviForm and Program admins. High-level steps are to create an app in Azure AD, create a group that contains CiviForm admins and finally update CiviForm server config to use the app and the group we just created. These instructions are just an example of how to get basic authentication working. Administrators are encouraged to adapt Azure AD part to fit their needs. 
+
+#### Configure Azure AD
+
+1. Login to Azure, go to "Active Active Directory" => "App Registrations" and create a new app. During creation set `Redirect URI` to be https://your-civiform-domain.gov/callback/AdClient replacing domain with your actual domain. The path `/callback/AdClient` is mandatory.
+   <details>
+    <summary>Screenshots</summary>
+
+    ![image](https://user-images.githubusercontent.com/252053/191863121-80b03cc5-dbfa-4929-b117-78c76142ceef.png)
+    ![image](https://user-images.githubusercontent.com/252053/191863133-cb396bb2-04cf-45a1-9d3a-1b805bcbab1c.png)
+
+   </details>
+2. Go to the newly created app, then Authentication and enable `ID tokens`. 
+   <details>
+    <summary>Screenshots</summary>
+  
+    ![image](https://user-images.githubusercontent.com/252053/191863376-d342092d-db5e-4111-96fa-a7fa751ed15f.png)
+   </details>
+3. Go to "Token configuration" section and add the following claims. Optional claims: `acct`, `email`. Groups claims: `Security groups`.
+   <details>
+    <summary>Screenshots</summary>
+  
+    ![image](https://user-images.githubusercontent.com/252053/191863628-6414d71b-c396-4628-8be6-6c58939f8f9b.png)
+    ![image](https://user-images.githubusercontent.com/252053/191863646-3d3c5e76-0510-48d2-9b78-bf27e18659e0.png)
+   </details>
+4. Go to "Certificates & secrets" and create a new secret that will be used by the CiviForm when it talks to Azure AD. Write down secret value somewhere temporarily. It will be used in a few steps. If you don't write it down and refresh the page - the value won't be accessible anymore, but you can always create a new secret. 
+   <details>
+    <summary>Screenshots</summary>     
+    
+    ![image](https://user-images.githubusercontent.com/252053/191864014-83385f7a-2c1b-4d29-8bbc-4506f6fd3631.png)
+   </details>
+5. Go to "Overview" and write down `Application (client) ID` and `OpenID Connect metadata document`. They will be used later fdf...
+
+   <details>
+    <summary>Screenshots</summary>     
+    
+    ![image](https://user-images.githubusercontent.com/252053/191864534-c79c78bd-effe-40b2-a22c-ecf6aa535698.png)
+   </details>
+6. We are done with setting up the Azure AD app. Now to go Azure "Groups" and create a new security group. That group will contain members that have CiviForm Admin access when they login to CiviForm. Other users, who are not members of that group, will be considered Program Admins. Once you created the group write down its ID it will be used later . 
```

Comment:
```suggestion
6. We are done with setting up the Azure AD app. Now to go Azure "Groups" and create a new security group. That group will contain members that have CiviForm Admin access when they login to CiviForm. Once you created the group write down its ID it will be used later . 
```